### ISSUE TITLE
Download weights for brightfield_nuclei

### ISSUE THREAD

**Issue by yliu-Acrivon** [instanseg/instanseg#141]
Hello!

I am a scientist from Acrivon Therapeutics. I am working a project to extract marker intensity from IHC images. Instanseg is very nice tool for nuclei detection.

I would like to fine tune Instanseg with my IHC images since it missed identifying some very bright tumor cells.

May you share the weights for brightfield_nuclei with me?

Any suggestions on the fine tune of Instanseg would be appreciated!

Best,

Spring

**Comment by alanocallaghan** [comment]
You could try changing the seed_threshold at inference if it's only missing some cells, without needing to retrain

**Comment by yliu-Acrivon** [comment]
Hi alanocallaghan! Thanks for the quick replay. Yes, I would try changing the seed_threshold. Still there are some other issues because images with various qualities. 

Could you share the weights for for brightfield_nuclei so that I can fine tune to see whether the performance improves?

Thank you!

**Comment by alanocallaghan** [comment]
I'm not that well versed in the technicalities but I think you can export the weights yourself, something like this:

```python
import instanseg
im = instanseg.InstanSeg() # brightfield is default
sd = im.instanseg.state_dict()
torch.save(sd, "weights.pth")
```

Would this be enough?

**Comment by yliu-Acrivon** [comment]
I just ran your code, and have got weights.pth of 15M. I guess that should be the right weights file for brightfield_nuclei, right?

**Comment by alanocallaghan** [comment]
I think so! You could obviously load the fluorescence model by specifying that model in `instanseg.InstanSeg()`. I think you should be able to fine-tune the model using those weights I think

**Comment by yliu-Acrivon** [comment]
I just asked Claude, the pth file generated from you code does not seem to be right. Please see below 

Ah — that's an important clarification. What you have is NOT the raw fine-tunable weights from the maintainers. It's a state_dict you extracted yourself from the TorchScript model.

This gives you a bag of weight tensors, but train.py expects a model folder with weights + config.

So this is NOT equivalent to the raw weights the maintainer offered to share.

**Comment by yliu-Acrivon** [comment]
Could you ask the Thibgolds on this too?

**Comment by alanocallaghan** [comment]
Perhaps you can ask Claude whether it is possible to convert from a state_dict to the weights format required for the model folder? The model "config"/definition is also stored within the instanseg object as far as I know.

I'm pretty sure this is a trivial problem given you have the model definition and weights but I'm not sure exactly what format you need and I have a strict deadline today for something else.

Thibaut and I no longer work in the same team; I'm sure he'll get to this when he has time.

**Comment by ThibautGoldsborough** [comment]
Hi, 

Sorry I should have replied earlier. Indeed converting the torchscript back to regular pytorch weights is not possible (at least not to my knowledge).  While I find a permanent solution to this, I'll link the [raw weights here ](https://drive.google.com/drive/folders/13_mvgK1co9dtUwEA1q5XISItu2pfbEXf?usp=drive_link). You should put them in the models folder and unzip them. Then  during training you can resume training using a smaller learning rate (e.g. ```python train.py -m_f brightfield_nuclei -lr 1e-4 -hotstart 0 -source [your_dataset]```

**Comment by yliu-Acrivon** [comment]
Hi Thibgolds, I am very glad that I got the raw weights now. Thank you so much. I might come back here if I am other questions when using them. Best.

**Comment by yliu-Acrivon** [comment]
Hello! Can you help with more questions regarding fine tuning or training InstanSeg? In your Github, you provided train.py and recommend that I need to incorporate your original dataset into the training. May I ask where to download the original data to the format that can be used by your code train.py?

Question 2: if I run train.py by adding our own data, how much computing resource I need? How long will it run for train.py on a Nvidia cuda system? Does the train.py totally train InstanSeg from fresh or just like a fine tuning of InstanSeg?

Question 3: Can I use some code developed by myself other than train.py to fine tune InstanSeg starting with the checkpoints you shared with me? Any issues to be taken into account if so?

Your help will be greatly appreciated! I do see that InstangSeg performs well on our data but need some fine tuning to be even better. It is a great tool!

Best regards,

**Comment by yliu-Acrivon** [comment]
Another question: I am using the brightfield_nuclei model, which of the InstanSeg raw data sets are important to be added together with my own data when using train.py? I remember you mentioned that to have CPDMI_2023 included when retrain fluorescence_nuclei_and_cells model.

Thank you!
